client-tools/Shell/V8ClientConnection.cpp

```diff
@@ -987,6 +987,99 @@ static void ClientConnection_httpPostRaw(
+                            fuzzer::RequestFuzzer* fuzzer) {
+  v8::HandleScope scope(isolate);
+  if (isExecutionDeadlineReached(isolate)) {
+    return;
+  }
+
+  TRI_V8_RETURN(v8connection->requestFuzz(isolate, fuzzer));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief ClientConnection method "fuzzRequests"
+////////////////////////////////////////////////////////////////////////////////
+
+static void ClientConnection_httpFuzzRequests(
+    v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+  if (isExecutionDeadlineReached(isolate)) {
+    return;
+  }
+
+  // get the connection
+  V8ClientConnection* v8connection = TRI_UnwrapClass<V8ClientConnection>(
+      args.Holder(), WRAP_TYPE_CONNECTION, TRI_IGETC);
+
+  if (v8connection == nullptr) {
+    TRI_V8_THROW_EXCEPTION_INTERNAL(
+        "fuzzRequests() must be invoked on an arango connection object "
+        "instance.");
+  }
+
+  // arg0 = number of iterations, arg1 = number of requests, arg2 = seed for
+  // rand
+  uint32_t numReqs = 200;
+  std::optional<uint32_t> numIts;
+  std::optional<uint32_t> seed;
+  if (args.Length() > 0) {
+    if (!args[0]->IsUint32()) {
+      TRI_V8_THROW_EXCEPTION_USAGE("<numRequests> must be an unsigned int.");
+    }
+    numReqs = TRI_ObjectToUInt64(isolate, args[0], false);
+  }
+  if (args.Length() > 1) {
+    if (!args[1]->IsUint32()) {
+      TRI_V8_THROW_EXCEPTION_USAGE("<numIts> must be an unsigned int.");
+    }
+    numIts = TRI_ObjectToUInt64(isolate, args[1], false);
+  }
+  if (args.Length() > 2) {
+    if (!args[2]->IsUint32()) {
+      TRI_V8_THROW_EXCEPTION_USAGE("<seed> must be an unsigned int.");
+    }
+    seed = TRI_ObjectToUInt64(isolate, args[2], false);
+  }
+  if (args.Length() > 3) {
+    TRI_V8_THROW_EXCEPTION_USAGE(
+        "fuzzRequests(<numRequests>, <numIterations>, <seed>). "
+        "(numIterations and numRequests must be > 0).");
+  }
+  auto fuzzer = std::make_unique<fuzzer::RequestFuzzer>(numIts, seed);
+  std::unordered_map<unsigned, uint32_t> fuzzReturnCodesCount;
+
+  fuzzer->logStart();
+  for (uint32_t i = 0; i < numReqs; ++i) {
+    httpFuzzRequest(v8connection, isolate, args, fuzzer.get());
+    velocypack::Builder builder;
+    TRI_V8ToVPack(isolate, builder, args.GetReturnValue().Get(), false);
+    fuzzReturnCodesCount[builder.slice().get("code").getUInt()]++;
+  }
+
+  std::string statistics =
+      "Fuzzer statistics:    Total requests=" + std::to_string(numReqs);
+  for (auto const& [returnCode, count] : fuzzReturnCodesCount) {
+    if (returnCode == 1000) {
+      statistics.append(",  Closed Connection=" + std::to_string(count));
+    } else {
+      statistics.append(",  " + std::to_string(returnCode) + "=" +
+                        std::to_string(count));
+    }
+  }
+
+  LOG_TOPIC("871a6", INFO, arangodb::Logger::COMMUNICATION) << statistics;
```

Comment:
I would prefer a V8 Object to be returned from this function, so that the caller/invoker can get hold of the returned statistics and do something with them programmatically.

Reply:
Addressed